doc/callbackinfo.md

```diff
@@ -0,0 +1,99 @@
+**WORK IN PROGRESS, NOT YET COMPLETE**
+
+# Callbackinfo
+
+The object representing the components of the JavaScript request being made.
+
+The Callbackinfo object is usually created and passed by the calling system.
+
+The Callbackinfo object contains the arguments passed by the caller. The number of arguments is returned by the `Length` method. Each individual argument can be accessed using the `operator[]` method.
+
+The `SetData` and `Data` methods are used to set and retrieve the data pointer contained in the Callbackinfo object.
+
+## Methods
+
+### Constructor
+
+```cpp
+CallbackInfo(napi_env env, napi_callback_info info);
+```
+
+- `[in] env`: The `napi_env` environment in which to construct the `CallbackInfo` object.
+- `[in] info`: The `napi_callback_info` data structure from which to construct the `CallbackInfo` object.
+
+### Env
+
+```cpp
+Napi::Env Env() const;
+```
+
+Returns the `Env` object in which the request is being made.
+
+### NewTarget
+
+```cpp
+Value NewTarget() const;
+```
+
+Returns the `new.target` value of the constructor call. If the CallbackInfo is not a constructor call, a call to `IsEmpty()` on the returned value returns true.
```

Comment:
This reads a bit strangely (to my ears).  Maybe:

If the function that was invoked (and for which the CallbackInfo was passed) is not a constructor call....

There is also similar wording in a few other places.